tests/conftest.py

```diff
@@ -2598,3 +2598,160 @@ def ceph_toolbox(request):
+        credentials
+        cloud_uls_factory: The cloud uls factory
+
+    """
+    created_ns_resources = []
+    created_ns_connections = []
+
+    def _create_ns_resources(amount=1, *args, **kwargs):
+        for i in range(amount):
+            # Create random connection_name and random namespace resource name
+            rand_ns_resource = create_unique_resource_name(constants.MCG_NS_RESOURCE, 'aws')
+            rand_connection = create_unique_resource_name(constants.MCG_NS_AWS_CONNECTION, 'aws')
+
+            # Create the actual namespace resource
+            target_bucket_name = mcg_obj.create_new_namespace_resource(
+                rand_ns_resource, rand_connection, constants.MCG_NS_REGION, cld_mgr, cloud_uls_factory)
+            check_ns_resource_validity(mcg_obj, rand_ns_resource,
+                                       target_bucket_name, constants.MCG_NS_AWS_ENDPOINT)
+
+            created_ns_resources.append(rand_ns_resource)
+            created_ns_connections.append(rand_connection)
+        return target_bucket_name, rand_ns_resource
+
+    def ns_resources_and_connections_cleanup():
+        for ns_resource in created_ns_resources:
+            delete_ns_resource(ns_resource, mcg_obj)
+
+        for ns_connection in created_ns_connections:
+            delete_ns_connection(ns_connection, mcg_obj)
+
+    request.addfinalizer(ns_resources_and_connections_cleanup)
+
+    return _create_ns_resources
+
+
+@pytest.fixture()
+def ns_bucket_factory(request, mcg_obj, cld_mgr):
+    """
+    Create a namespace bucket factory. Calling this fixture creates a new namespace bucket(s).
+    For a custom amount, provide the 'amount' parameter.
+
+    Args:
+        mcg_obj (MCG): An MCG object
+        cld_mgr (CM): A cloud manager object containing the MCG S3 connection
+        credentials
+
+    """
+    created_ns_buckets = []
+
+    def _create_ns_buckets(amount=1, write_resource=None, read_resources=None, *args, **kwargs):
+        if write_resource is None:
+            write_resource = []
+        if read_resources is None:
+            read_resources = [[]]
+        for i in range(amount):
+            # Create random namespace bucket name
+            rand_ns_bucket = create_unique_resource_name('aws', constants.MCG_NS_BUCKET)
+
+            # Create actual namespace bucket
+            mcg_obj.create_new_namespace_bucket(rand_ns_bucket, write_resource[i], read_resources[i])
+            created_ns_buckets.append(rand_ns_bucket)
+            check_ns_bucket_validity(mcg_obj, rand_ns_bucket, write_resource[i], read_resources[i])
+        return rand_ns_bucket
+
+    def ns_buckets_cleanup():
+        for ns_bucket in created_ns_buckets:
+            delete_ns_bucket(ns_bucket, mcg_obj)
+
+    request.addfinalizer(ns_buckets_cleanup)
+
+    return _create_ns_buckets
+
+
+def check_ns_resource_validity(mcg_obj, ns_resource_name, target_bucket_name, endpoint):
```

Comment:
This should not be in conftest. I suggest to move it to `ocs_ci/ocs/resources/objectbucket.py`.

Reply:
since it doesn't handle the bucket I moved it to mcg.py and not to objectbucket.py, is that ok?